### PR TITLE
Only replace strings in text files

### PR DESCRIPTION
**Pull request by carlo-quinonez**
Check the content type starts with "text" before doing any replacements.

### CODE REVIEW

File: lib/stringReplaceMiddleware.js [hunk 1]
```diff
@@ -3,14 +3,19 @@ const stringReplaceStream = require('./stringReplaceStream');
 
 module.exports = (replacements) => (req, res, next) => {
```
Comment (HugoPoi) [inline lib/stringReplaceMiddleware.js]: add `options` parameter here like `module.exports = (replacements, options) =>`

File: lib/stringReplaceMiddleware.js [hunk 2]
```diff
@@ -3,14 +3,19 @@ const stringReplaceStream = require('./stringReplaceStream');
...
-            return next(err);
+        const contentType = res.get('content-type') || "";
+        if (contentType.startsWith('text')) {
+            if (err) {
+                res.unhijack(); // Make the original res object work again
+                return next(err);
+            }
+            res.removeHeader('Content-Length');
+            res
+                .pipe(stringReplaceStream(replacements))
```
Comment (HugoPoi) [inline lib/stringReplaceMiddleware.js:18]: pass `options` here `.pipe(stringReplaceStream(replacements, options))
`

File: lib/stringReplaceMiddleware.js [hunk 3]
```diff
@@ -3,14 +3,19 @@ const stringReplaceStream = require('./stringReplaceStream');
 
 module.exports = (replacements) => (req, res, next) => {
     hijackResponse(res, function (err, res) {
-        if (err) {
-            res.unhijack(); // Make the original res object work again
-            return next(err);
+        const contentType = res.get('content-type') || "";
+        if (contentType.startsWith('text')) {
```
Comment (HugoPoi) [inline lib/stringReplaceMiddleware.js]: will not work for `application/json` contentType
`options.contentTypeFilterRegexp.test(contentType)`
With a default value for `options.contentTypeFilterRegexp`
`options.contentTypeFilterRegexp = /^text\/|^application\/json$|^application\/xml$/` maybe

**Review by HugoPoi — COMMENTED** [review]
I have done a little review of your PR,
I think adding `options.contentTypeFilterRegexp` is a must have.

**Comment by bfncs** [comment]
Thanks for submitting the PR, @carlo-quinonez.

You solution is great in general, please use a configurable regex on the content type to decide whether or not to apply the filter like @HugoPoi suggested. Please add this and I will happily merge.

Also there are currently no tests for the module at all, it was just a quick prototype and I wasn't aware anyone is using it except me. I'll think an adequate testing strategy in the next days, a small number of integration tests should suffice.

**Comment by carlo-quinonez** [comment]
I updated the PR according to the comments.

File: lib/stringReplaceMiddleware.js (hunk 3 above)
Comment (carlo-quinonez) [reply lib/stringReplaceMiddleware.js]: I added the regex provided as the defauit value.

**Comment by carlo-quinonez** [comment]
I updated the README as well to reflect the new configuration options.

File: lib/stringReplaceMiddleware.js [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
-module.exports = (replacements) => (req, res, next) => {
+const defaultOptions = {
+  contentTypeFilterRegexp: /^text\/|^application\/json$|^application\/xml$/,
+}
+
+module.exports = (replacements, options = defaultOptions ) => (req, res, next) => {
     hijackResponse(res, function (err, res) {
         const contentType = res.get('content-type') || "";
-        if (contentType.startsWith('text')) {
+        if (contentType.match(options.contentTypeFilterRegexp)) {
```
Comment (HugoPoi) [inline lib/stringReplaceMiddleware.js]: Better to do
```javascript
if (options.contentTypeFilterRegexp.test(contentType)) {
```

Comment (luzlab) [reply lib/stringReplaceMiddleware.js]: Nice! I didn't like that I needed a default value of `""` for contentType. I made the change you suggested and removed the default value.

**Comment by carlo-quinonez** [comment]
Can we merge this?

**Comment by carlo-quinonez** [comment]
@bfncs Merge?

**Comment by bfncs** [comment]
The PR looks great, thanks for the contribution. This completely slipped my attention, sorry for having you waiting here.

**Pull request merged**